File: LeetCode/Basic/1431/main.go
package solution

func kidsWithCandies(candies []int, extraCandies int) []bool {
	var max int
	for _, candy := range candies {
		if candy >= max {
			max = candy
		}
	}
	ans := make([]bool, len(candies))
	for i, candy := range candies {
		if candy+extraCandies >= max {
			ans[i] = true
		} else {
			ans[i] = false
		}
	}
	return ans
}

File: LeetCode/Basic/1431/main.py
from typing import List


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max = 0
        for candy in candies:
            if max < candy:
                max = candy

        ans = [False]*len(candies)
        for i, candy in enumerate(candies):
            if candy + extraCandies >= max:
                ans[i] = True
            else:
                ans[i] = False
        return ans
